src/com/esotericsoftware/kryo/io/Output.java

```diff
@@ -274,6 +273,17 @@ public void writeBytes (byte[] bytes, int offset, int count) throws KryoExceptio
 		}
 	}
 
+	/** Writes count bytes from long, the last byte written is the lowest byte from the long.
+	 *  Note the number of bytes is not written. */
+	public void writeBytesFromLong (long bytes, int count) {
+		require(count);
+		int p = position;
+		position = p + count;
+		for (int i = count - 1; i >= 0; i--) {
+			buffer[p++] = (byte) (bytes >> (i << 3));

```

Comment:
Writing could be one less shift per iteration (untested):
```
for (int i = (count - 1) << 3; i >= 0; i -= 8)
	buffer[p++] = (byte)(bytes >> i);
```
Same for ByteBufferOutput.

Reading could match, like:
```
for (int i = (count - 1) << 3; i > 0; i -= 8)
	bytes |= (buffer[p++] & 0xFF) << i;
```
However, for reading what you have may be better: simpler initializer and similar operations per iteration.

Reply:
I tried to do that, but the read side got a bit more complex:
```java
long bytes = buffer[p] >= 0 ? 0 : ~(-1L >>> ((8-count) << 3));
for (int i = (count - 1) << 3; i >= 0; i -= 8) {
	bytes |= (buffer[p++] & (long) 0xFF) << i;
}
```
And also much slower (while the write side stayed the same when it comes to speed):
![new-loops](https://github.com/EsotericSoftware/kryo/assets/4157004/b43a5d8b-76ac-4dd6-818c-cd711ca8ecd4)

So in 813ae4b I did something different - I unrolled the loops manually replacing it with some switches and repeated code. This got the reading/writing a little bit faster:

![unrolled-loops](https://github.com/EsotericSoftware/kryo/assets/4157004/f8ed2e76-6d9a-4db3-97dc-74e26ae13060)